applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp

```diff
@@ -71,32 +71,14 @@ int UPwSmallStrainElement<TDim, TNumNodes>::Check(const ProcessInfo& rCurrentPro
         check_properties.CheckPermeabilityProperties(r_geometry.WorkingSpaceDimension());
     }
 
-    // Verify that the constitutive law exists
-    KRATOS_ERROR_IF_NOT(this->GetProperties().Has(CONSTITUTIVE_LAW))
-        << "Constitutive law not provided for property " << this->GetProperties().Id() << std::endl;
-
-    // Verify that the constitutive law has the correct dimension
-    const SizeType strainSize = this->GetProperties().GetValue(CONSTITUTIVE_LAW)->GetStrainSize();
-    if (TDim == 2) {
-        KRATOS_ERROR_IF_NOT(strainSize == VOIGT_SIZE_2D_PLANE_STRAIN)
-            << "Wrong constitutive law used. This is a 2D element! expected "
-               "strain size is "
-            << VOIGT_SIZE_2D_PLANE_STRAIN << " But received: " << strainSize
-            << " in element id: " << this->Id() << std::endl;
-    } else {
-        KRATOS_ERROR_IF_NOT(strainSize == VOIGT_SIZE_3D)
-            << "Wrong constitutive law used. This is a 3D element! expected "
-               "strain size is "
-            << VOIGT_SIZE_3D << " But received: " << strainSize << " in element id: " << this->Id()
-            << std::endl;
-    }
+    check_properties.CheckAvailabilityAndSpecified(CONSTITUTIVE_LAW);
+    const auto expected_sizes = (TDim == 2 ? std::vector<std::size_t>{VOIGT_SIZE_2D_PLANE_STRAIN}
+                                           : std::vector<std::size_t>{VOIGT_SIZE_3D});
+    ConstitutiveLawUtilities::CheckStrainSize(r_properties, expected_sizes, TDim, this->Id());
```

Comment:
In this refactor, the functionality is kept exactly as is, which is in general indeed a good idea! In this case, to avoid having to provide a list of allowed expected sizes (which is a bit counterintuitive) and simplify the code, we could get the expected size from the stress state policy (as Wijtze Pieter also mentioned in one of the comments):
```suggestion
    const auto expected_size = GetStressStatePolicy().GetVoigtSize();
    ConstitutiveLawUtilities::CheckStrainSize(r_properties, expected_size, TDim, this->Id());
```
This way we check whether the constitutive law is consistent with the element itself, we don't need to recompute the expected value based on dimension and we only check for a single value, not a list of allowed values. 

The same then holds for the other places where `ConstitutiveLawUtilities::CheckStrainSize` and of course the function itself should be slightly adjusted to obtain the single value instead of the list

Reply:
Thank you. Following Wijtze Pieter and your suggestion, I have simplified the function.